stock_release_channel_shipment_lead_time/models/stock_release_channel.py

```diff
@@ -49,7 +51,13 @@ def _add_shipment_lead_time(self, dt):
         if not self.delivery_calendar_id:
             shipment_tz = fields.Datetime.add(dt_tz, days=self.shipment_lead_time)
         else:
-            days = self.shipment_lead_time + 1
+            days = self.shipment_lead_time
+            if days:
+                # We consider the current day as valid, we add a day, and then
+                # we look for open days for the given delay. Reset time to 0 in
+                # case the end time if after the warehouse calendar end time.
+                dt_tz += timedelta(days=1)
+                dt_tz = dt_tz.replace(hour=0, minute=0, second=0, microsecond=0)
```

Comment:
Still with the Odoo API here:
```suggestion
                dt_tz = fields.Datetime.start_of(dt_tz, "day")
```

Reply:
This one is not equivalent
```
(Pdb) p fields.Datetime.start_of(dt_tz, "day")
datetime.datetime(2025, 1, 2, 0, 0)
(Pdb) p dt_tz.replace(hour=0, minute=0, second=0, microsecond=0)
datetime.datetime(2025, 1, 2, 0, 0, tzinfo=<DstTzInfo 'Europe/Brussels' CET+1:00:00 STD>)
```